extension.js

```diff
@@ -7,6 +7,22 @@ const PanelMenu = imports.ui.panelMenu;
 const Me = ExtensionUtils.getCurrentExtension();
 const workspaceManager = global.workspace_manager;
 
+
+// workspace switch to previous / next
+let scroll_wrap = false;
+function workspace_switch(d){
```

Comment:
I'd prefer if `scrall_wrap` was a parameter to this function rather than a global variable.

Global variables introduce lots of state which makes it harder to maintain in the long term :)

Reply:
got it, I'll change it like `change_on_click` does

Reply:
I also thinking about whether global function `workspace_switch(d)` should exist

Maybe attach it as a property / method of workspaceManager or WorkspaceLayout

Reply:
This is good enough as it is I think :) Must important thing is that it works and is easy to read which I think we now have achieved.